### PR TITLE
Modal invoice through btcpay.js

### PR DESCRIPTION
**Pull request by rockstardev**

### CODE REVIEW

**Comment by NicolasDorier** [comment]
coooooool.

Though, how do I use it?

File: BTCPayServer/wwwroot/modal/btcpay.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    function showInvoice(invoiceId, params) {
+        window.document.body.appendChild(iframe);
+        var invoiceUrl = origin + '/invoice?id=' + invoiceId + '&v=2&view=modal';
+        if (params && params.animateEntrance === false) {
+            invoiceUrl += '&animateEntrance=false';
+        }
+        iframe.src = invoiceUrl;
+    }
+
+    function setApiUrlPrefix(urlPrefix) {
```
Comment (NicolasDorier) [inline BTCPayServer/wwwroot/modal/btcpay.js]: can you verify that the user does not set a `/` at the end by mistake? this is classical foot shooting.

**Comment by rockstardev** [comment]
@NicolasDorier I have left `test.html` in same folder as example. Since we are _decentralized_ we won't have `btcpay.js` hosted at one place so setup needs to both reference `.js` file and then initialize site root for calls to show invoices, see: https://github.com/btcpayserver/btcpayserver/blob/364bffcda310b3ba214d263c6e138a9f6cae84db/BTCPayServer/wwwroot/modal/test.html#L5

I want to combine this with pay button so that users can create invoice when clicking button and then show that invoice... started conversation in Slack on stuff I need to make that happen.

**Comment by NicolasDorier** [comment]
@rockstardev can we expose this test page inside BTCPay somewhere? Because how does the user can discover this feature?

**Comment by NicolasDorier** [comment]
or better, just documenting somewhere

File: BTCPayServer/wwwroot/modal/test.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿<html>
+<head>
+    <script src="btcpay.js"></script>
+    <script type="text/javascript">
+        btcpay.setApiUrlPrefix("http://127.0.0.1:14142");
```
Comment (NicolasDorier) [inline BTCPayServer/wwwroot/modal/test.html]: because NBXPlorer has the serverUrl, we don't have to use 127.0.0.1.

**Comment by NicolasDorier** [comment]
Merging it, but @rockstardev can you delete the test.html, and use this popup in the Invoices list.

I am thinking this, in the invoice list, change the `Checkout` linke to  `Checkout [^]`. And if the user click on `^` we should the modal.

**Pull request merged**